modules/linear_boltzmann_solvers/discrete_ordinates_solver/iterative_methods/sweep_wgs_context.cc

```diff
@@ -136,7 +135,7 @@ SweepWGSContext::PostSolveCallback()
   // currently used in OpenSn). This step also zeros out balance variables and computes the correct
   // in-flow and out-flow. Classic Richardson calls this solely to compute balance quantities (note
   // that it does cost us an extra sweep that is technically not necessary).
-  lbs_ss_solver.ZeroOutflowBalanceVars(groupset);
+  dynamic_cast<DiscreteOrdinatesSolver&>(lbs_solver).ZeroOutflowBalanceVars(groupset);
```

Comment:
just curious why the dynamic cast here? I am always suspicious of something underlying. can we add a virtual function ZeroOutflowBalanceVars in the base class so that we don't need the dynamic_cast?

Reply:
Because the underlying type of `lbs_solver` is `LBSSolver` and not `DiscreteOrdinatesSolver`. Prior to this PR, this context stored an `lbs_solver` of type `LBSSolver&` and `lbs_ss_solver` of type `DiscreteOrdinatesSolver&` (a reference to the same object but cast to each respective type. We can certainly add a virtual method to `LBSSolver`, but I hesitate to go down that path because `LBSSolver` is already so massive and we plan to clean it up.